### PR TITLE
docs: Remove mention of mailing list

### PR DESCRIPTION
**Pull request by nzakas**
Removes all mentions of the now-retired mailing list in our documentation.

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Because we've retired the mailing list, I've removed all mention of it.

### CODE REVIEW

File: packages/eslint-config-eslint/README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ In your `.eslintrc` file, add:
 
 ### Where to ask for help?
 
-Join our [Mailing List](https://groups.google.com/group/eslint) or [Chatroom](https://eslint.org/chat)
+Open a [discussion](https://github.com/eslint/eslint/discussions) or stop by our [Discord server](https://eslint.org/chat) instead of filing an issue.
```
Comment (mdjermanovic) [inline packages/eslint-config-eslint/README.md:35]: In this file, there's another one here:

https://github.com/eslint/eslint/blob/9b2fcf7e928fc92ac6d43617bdee1bda250b7491/packages/eslint-config-eslint/README.md?plain=1#L5

**Comment by mdjermanovic** [comment]
Can we also remove this one?

https://github.com/eslint/eslint/blob/9b2fcf7e928fc92ac6d43617bdee1bda250b7491/docs/src/_data/links.json#L5

File: docs/src/maintain/manage-issues.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ There are four primary issue categories:
 1. **Bug** - something isn't working the way it's expected to work.
 1. **Enhancement** - a change to something that already exists. For instance, adding a new option to an existing rule or a bug in a rule where fixing it will result in the rule reporting more problems (in this case, use both "Bug" and "Enhancement").
 1. **Feature** - adding something that doesn't already exist. For example, adding a new rule, new formatter, or new command line flag.
-1. **Question** - an inquiry about how something works that won't result in a code change. We'd prefer if people use the mailing list or chatroom for questions, but sometimes they'll open an issue.
+1. **Question** - an inquiry about how something works that won't result in a code change. We'd prefer if people use discussions or chatroom for questions, but sometimes they'll open an issue.
```
Comment (harish-sethuraman) [inline docs/src/maintain/manage-issues.md:27]: should we mention github discussions? or is it fine?

File: packages/eslint-config-eslint/README.md (hunk 1 above)
Comment (amareshsm) [reply packages/eslint-config-eslint/README.md:35]: In `links.json` file, there is one occurrence.
https://github.com/eslint/eslint/blob/337f7ed96131d873be7ae6b010739476d0ad15e9/docs/src/_data/links.json#L5

**Comment by nzakas** [comment]
I was just doing this as a first pass. :) I'd love it if someone else could finish up this work so I can focus on RFCs. You can clone the branch I used for this PR and just modify it directly.

**Pull request closed without merging**